### PR TITLE
Implement `get_training_dataset` method in `DatasetClient`

### PR DESCRIPTION
**Pull request by ljcornel**
The DatasetClient now provides a method to get a list of images and video frames used in a specified dataset split (training/testing/validation).

### CODE REVIEW

File: geti_sdk/rest_clients/dataset_client.py [hunk 1]
```diff
@@ -157,3 +159,90 @@ def _media_folder_has_dataset_subfolders(self, path_to_folder: str) -> bool:
...
+
+    def get_training_dataset_summary(self, model: Model) -> Dict[str, Any]:
+        """
+        Return information concerning the training dataset for the `model`.
+        This includes the number of images and video frames, and the statistics for
+        the subset splitting (i.e. the number of training, test and validation
+        images/video frames)
+
+        :param model: Model to get the training dataset for
+        :return: Dictionary containing the training dataset info for the model
```
Comment (igor-davidyuk) [inline geti_sdk/rest_clients/dataset_client.py:171]:
```suggestion
        :return: Dictionary containing the training dataset info for the model
        :raises: ValueError if unable to fetch the required dataset information from the model
```

File: geti_sdk/rest_clients/dataset_client.py [hunk 2]
```diff
@@ -157,3 +159,90 @@ def _media_folder_has_dataset_subfolders(self, path_to_folder: str) -> bool:
...
+                        frames_response = self.session.get_rest_response(
+                            url=frame_page, method="POST", data=post_data
+                        )
+                        for frame in frames_response["media"]:
+                            frame["video_id"] = video_id
+                            video_frame = MediaRESTConverter.from_dict(
+                                frame, VideoFrame
+                            )
+                            video_frames.append(video_frame)
+                        frame_page = frames_response.get("next_page", None)
```
Comment (igor-davidyuk) [inline geti_sdk/rest_clients/dataset_client.py:246]: should it be `next_page`?

File: geti_sdk/rest_clients/dataset_client.py (hunk 1 above)
Comment (ljcornel) [reply geti_sdk/rest_clients/dataset_client.py]: Added

File: geti_sdk/rest_clients/dataset_client.py (hunk 2 above)
Comment (ljcornel) [reply geti_sdk/rest_clients/dataset_client.py]: `next_page` is already in use, changed it to `next_frame_page`

File: geti_sdk/rest_clients/dataset_client.py [hunk 3]
```diff
@@ -157,3 +159,90 @@ def _media_folder_has_dataset_subfolders(self, path_to_folder: str) -> bool:
...
+            )
+        training_dataset = self.session.get_rest_response(
+            url=f"{self.base_url}/{dataset_storage_id}/training_revisions/{revision_id}",
+            method="GET",
+        )
+        return training_dataset
+
+    def get_media_in_training_dataset(
+        self, model: Model, subset: str = "training"
+    ) -> Tuple[MediaList[Image], MediaList[VideoFrame]]:
```
[the remark this answers is not in the document]
Comment (ljcornel) [reply geti_sdk/rest_clients/dataset_client.py]: Agreed, created a new class `Subset` which contains both lists

File: geti_sdk/rest_clients/dataset_client.py [hunk 4]
```diff
@@ -157,3 +159,90 @@ def _media_folder_has_dataset_subfolders(self, path_to_folder: str) -> bool:
...
+        if dataset_storage_id is None or revision_id is None:
+            raise ValueError(
+                f"Unable to fetch the required dataset information from the model. "
+                f"Expected dataset and revision id's, got {ds_info} instead."
+            )
+        training_dataset = self.session.get_rest_response(
+            url=f"{self.base_url}/{dataset_storage_id}/training_revisions/{revision_id}",
+            method="GET",
+        )
+        return training_dataset
```
[the remark this answers is not in the document]
Comment (ljcornel) [reply geti_sdk/rest_clients/dataset_client.py]: Good point, added it

**Review by igor-davidyuk — APPROVED** [review]

**Pull request merged**